### PR TITLE
feat: Adds azure support for backup snapshot export bucket

### PR DESCRIPTION
**Pull request by oarbusi**
## Description

Adds Azure support for backup snapshot export bucket
This is a copy of https://github.com/mongodb/terraform-provider-mongodbatlas/pull/2471 done by @lulizv to be able to run acceptance test

## Type of change:

- [ ] Bug fix (non-breaking change which fixes an issue). Please, add the "bug" label to the PR.
- [x] New feature (non-breaking change which adds functionality). Please, add the "enhancement" label to the PR. A migration guide must be created or updated if the new feature will go in a major version.
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected). Please, add the "breaking change" label to the PR. A migration guide must be created or updated.
- [ ] This change requires a documentation update
- [ ] Documentation fix/enhancement

## Required Checklist:

- [x] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [x] I have read the [contributing guides](https://github.com/mongodb/terraform-provider-mongodbatlas/blob/master/contributing/README.md)
- [ ] I have checked that this change does not generate any credentials and that **they are NOT accidentally logged anywhere**.
- [x] I have added tests that prove my fix is effective or that my feature works per HashiCorp requirements
- [ ] I have added any necessary documentation (if appropriate)
- [x) I have run make fmt and formatted my code
- [ ] If changes include deprecations or removals, I defined an isolated PR with a relevant title as it will be used in the auto-generated changelog.
- [ ] If changes include removal or addition of 3rd party GitHub actions, I updated our internal document. Reach out to the APIx Integration slack channel to get access to the internal document.

### CODE REVIEW

**Review by kyuan-mongodb — APPROVED** [review]
LGTM!

File: examples/mongodbatlas_cloud_backup_snapshot_export_bucket/azure/main.tf [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+  azure {
+    atlas_azure_app_id   = var.azure_atlas_app_id
+    service_principal_id = azuread_service_principal.mongo.id
+    tenant_id            = var.tenant_id
+  }
+}
+
+
+resource "mongodbatlas_cloud_backup_snapshot_export_bucket" "test" {
```
Comment (EspenAlbert) [inline examples/mongodbatlas_cloud_backup_snapshot_export_bucket/azure/main.tf:23]: Can we add `README.md` files to these examples?

File: internal/service/cloudbackupsnapshotexportbucket/resource_cloud_backup_snapshot_export_bucket_test.go [hunk 2]
```diff
@@ -69,6 +73,65 @@ func basicTestCase(tb testing.TB) *resource.TestCase {
...
+			acc.PreCheckAzureExportBucket(t)
+		},
+		ProtoV6ProviderFactories: acc.TestAccProviderV6Factories,
+		CheckDestroy:             checkDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: configAzureBasic(projectID, atlasAzureAppID, servicePrincipalID, tenantID, bucketName, serviceURL),
+				Check: resource.ComposeAggregateTestCheckFunc(
+					checkExists(resourceName),
+					resource.TestCheckResourceAttr(resourceName, "project_id", projectID),
```
Comment (EspenAlbert) [inline internal/service/cloudbackupsnapshotexportbucket/resource_cloud_backup_snapshot_export_bucket_test.go]: Can we use `acc.AddAttrChecks` to remove duplication?

File: internal/service/cloudbackupsnapshotexportbucket/resource_cloud_backup_snapshot_export_bucket_test.go [hunk 3]
```diff
@@ -69,6 +73,65 @@ func basicTestCase(tb testing.TB) *resource.TestCase {
...
+					resource.TestCheckResourceAttr(dataSourcePluralName, "results.0.cloud_provider", "AZURE"),
+					resource.TestCheckResourceAttrSet(dataSourcePluralName, "results.0.role_id"),
+				),
+			},
+			{
+				ResourceName:            resourceName,
+				ImportStateIdFunc:       importStateIDFunc(resourceName),
+				ImportState:             true,
+				ImportStateVerify:       true,
+				ImportStateVerifyIgnore: []string{"container_id"},
```
Comment (EspenAlbert) [inline internal/service/cloudbackupsnapshotexportbucket/resource_cloud_backup_snapshot_export_bucket_test.go]: why is this ignored?

**Review by EspenAlbert — APPROVED** [review]
Great job! Only a few minor comments.

File: internal/service/cloudbackupsnapshotexportbucket/resource_cloud_backup_snapshot_export_bucket_test.go (hunk 3 above)
Comment (oarbusi) [reply internal/service/cloudbackupsnapshotexportbucket/resource_cloud_backup_snapshot_export_bucket_test.go]: Great catch! It wasn't needed, so I removed it

File: internal/service/cloudbackupsnapshotexportbucket/resource_cloud_backup_snapshot_export_bucket_test.go (hunk 2 above)
Comment (oarbusi) [reply internal/service/cloudbackupsnapshotexportbucket/resource_cloud_backup_snapshot_export_bucket_test.go]: Done

File: examples/mongodbatlas_cloud_backup_snapshot_export_bucket/azure/main.tf (hunk 1 above)
Comment (oarbusi) [reply examples/mongodbatlas_cloud_backup_snapshot_export_bucket/azure/main.tf:23]: Done

**Review by AgustinBettati — APPROVED** [review]
LGTM

**Comment by edzell1** [comment]
@oarbusi - thanks to you and the team for a quick turnaround!  Do you know when a provider release with this change will happen? Thanks, Alan

**Comment by oarbusi** [comment]
Hi @edzell1 , 
We are planning to release v1.18.0 next week. Let me know if you have any other concern.

Thanks,

**Pull request merged**